verticapy/performance/vertica/collection/profile_import.py

```diff
@@ -54,33 +59,97 @@ def __init__(
         filename: str
             The tarball of parquet files to load into the database.
         """
-        # TODO: at this time, we don't run any checks automatically
-        # during __init__. Instead, we run the chceks manually. Running
-        # the checks manually facilitates early testing.
 
         self.target_schema = target_schema
         self.key = key
         self.filename = filename
-        self.input_file_obj = None
-        self.skip_create_table = skip_create_table
         self.logger = logging.getLogger("ProfileImport")
 
+        # initialize internal attributes
+        self.tarfile_obj = None
+        self.bundle_version = None
+        self.raise_when_missing_files = False
+        self.skip_create_table = False
+        self.tmp_path = os.getcwd()
+
+    @property
+    def skip_create_table(self) -> bool:
+        """
+        ProfileImport will SKIP running ddl statements to create new
+        tables when skip_creat_table is True. Otherwise it will run
+        DDL statements. Default value is False.
+
+        The DDL statements are:
+           CREATE TABLE ... IF NOT EXISTS
+           CREATE PROJECTION ... IF NOT EXISTS
+        """
+        return self._skip_create_table
+
+    @skip_create_table.setter
+    def skip_create_table(self, val: bool) -> None:
+        if not isinstance(val, bool):
+            raise ValueError(
```

Comment:
I think it is better to raise `TypeError` here instead of `ValueError`.

Reply:
+1 TypeError is better, I will change this.

Reply:
66219ca2 and added a test